src/convertToObject.js

```diff
@@ -3,10 +3,32 @@
 /**
  * @param {string} sourceString
  *
- * @return {object}
+ * @return {object}n
  */
-function convertToObject(sourceString) {
-  // write your code here
-}
+function convertToObject(stylesString) {
+  // Создаем копию параметра, чтобы не изменять его
+  const modifiedStylesString = stylesString
+    .replace(/;+\s*$/, '') // Удаляет точки с запятой в конце строки
+    .trim();
+
+  const stylesObject = {};
+
+  // Разделяем строки на отдельные правила
+  const rules = modifiedStylesString.split(/;\s*/);
+
+  rules.forEach((rule) => {
+    // Пропускаем пустые строки
+    if (rule.trim() === '') {
+      return;
+    }
 
+    const [property, value] = rule.split(/:\s*/);
```

Comment:
When splitting the rule into property and value, you should consider edge cases where the property or value might contain ':' characters. This would split the string incorrectly. Use a limit in your `split` call to ensure that only the first ':' character is used for splitting.